### PR TITLE
Manage multi storeID on a token and add feed key distinct to token (44802 & 46128)

### PR DESCRIPTION
**Pull request by bogdan202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Adding the possibility configure storeID + using feed_key 
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes 44802 46128
| How to test?  | Fill in the registration form

### CODE REVIEW

File: controllers/admin/AdminShoppingfeedAccountSettings.php [hunk 1]
```diff
@@ -319,6 +352,7 @@ public function postProcess()
     public function saveToken($shop_id, $lang_id, $currency_id)
     {
         $token = Tools::getValue(Shoppingfeed::AUTH_TOKEN);
+        $store_id = Tools::getValue('store_id', '');
```
Comment (clotairer) [inline controllers/admin/AdminShoppingfeedAccountSettings.php:355]: store_id should be an integer according to the object model definition.
You probably should cast it or add a validation before using it.

File: controllers/admin/AdminShoppingfeedAccountSettings.php [hunk 2]
```diff
@@ -363,6 +402,7 @@ public function login($shop_id, $lang_id, $currency_id)
     {
         $username = Tools::getValue('username');
         $password = Tools::getValue('password');
+        $store_id = Tools::getValue('store_id');
```
Comment (clotairer) [inline controllers/admin/AdminShoppingfeedAccountSettings.php:405]: store_id should be an integer according to the object model definition.
You probably should cast it or add a validation before using it.

**Review by clotairer — CHANGES_REQUESTED** [review]

File: controllers/admin/AdminShoppingfeedAccountSettings.php (hunk 2 above)
Comment (bogdan202) [reply controllers/admin/AdminShoppingfeedAccountSettings.php]: the modifications were added

File: controllers/admin/AdminShoppingfeedAccountSettings.php (hunk 1 above)
Comment (bogdan202) [reply controllers/admin/AdminShoppingfeedAccountSettings.php]: the modifications were added

**Review by clotairer — APPROVED** [review]

**Pull request merged**